### PR TITLE
8329083: RISC-V: Update profiles supported on riscv

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you help to review this patch to update vm flags related to riscv profile?
Thanks

Currently there are vm options like -XX:+UseRVA20U64 and -XX:+UseRVA22U64 on riscv to indicate the supported riscv extension via profiles.
These profiles should be updated to reflect the full supported extensions and new profile like UseRVA23U64 should be added too.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329083](https://bugs.openjdk.org/browse/JDK-8329083): RISC-V: Update profiles supported on riscv (**Enhancement** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18599/head:pull/18599` \
`$ git checkout pull/18599`

Update a local copy of the PR: \
`$ git checkout pull/18599` \
`$ git pull https://git.openjdk.org/jdk.git pull/18599/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18599`

View PR using the GUI difftool: \
`$ git pr show -t 18599`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18599.diff">https://git.openjdk.org/jdk/pull/18599.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18599#issuecomment-2034193165)

### CODE REVIEW

File: src/hotspot/cpu/riscv/vm_version_riscv.hpp [hunk 1]
```diff
@@ -170,6 +170,56 @@ class VM_Version : public Abstract_VM_Version {
...
+  // https://github.com/riscv/riscv-profiles/blob/main/profiles.adoc#rva20-profiles
+  #define RV_USE_RVA20U64                            \
+    RV_ENABLE_EXTENSION(UseRVC)                      \
+
+  static void useRVA20U64Profile();
+
+  // https://github.com/riscv/riscv-profiles/blob/main/profiles.adoc#rva22-profiles
+  #define RV_USE_RVA22U64                            \
+    RV_ENABLE_EXTENSION(UseRVC)                      \
+    RV_ENABLE_EXTENSION(UseRVV)                      \
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/vm_version_riscv.hpp:191]: Note that V was optional in RVA22U64.

File: src/hotspot/cpu/riscv/vm_version_riscv.hpp [hunk 2]
```diff
@@ -170,6 +170,56 @@ class VM_Version : public Abstract_VM_Version {
...
+    RV_ENABLE_EXTENSION(UseZicboz)                   \
+    RV_ENABLE_EXTENSION(UseZihintpause)              \
+
+  static void useRVA22U64Profile();
+
+  // https://github.com/riscv/riscv-profiles/blob/main/rva23-profile.adoc#rva23u64-profile
+  #define RV_USE_RVA23U64                           \
+    RV_ENABLE_EXTENSION(UseRVC)                     \
+    RV_ENABLE_EXTENSION(UseRVV)                     \
+    RV_ENABLE_EXTENSION(UseZacas)                   \
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/vm_version_riscv.hpp:208]: Zacas is RVA23U64 Optional Extension.

File: src/hotspot/cpu/riscv/vm_version_riscv.hpp [hunk 3]
```diff
@@ -170,6 +170,56 @@ class VM_Version : public Abstract_VM_Version {
...
+    RV_ENABLE_EXTENSION(UseZbb)                     \
+    RV_ENABLE_EXTENSION(UseZbs)                     \
+    RV_ENABLE_EXTENSION(UseZfh)                     \
+    RV_ENABLE_EXTENSION(UseZcb)                     \
+    RV_ENABLE_EXTENSION(UseZic64b)                  \
+    RV_ENABLE_EXTENSION(UseZicbom)                  \
+    RV_ENABLE_EXTENSION(UseZicbop)                  \
+    RV_ENABLE_EXTENSION(UseZicboz)                  \
+    RV_ENABLE_EXTENSION(UseZihintpause)             \
+    RV_ENABLE_EXTENSION(UseZvfh)                    \
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/vm_version_riscv.hpp:219]: Zvfh is RVA23U64 Optional Extension too.

File: src/hotspot/cpu/riscv/globals_riscv.hpp [hunk 4]
```diff
@@ -101,6 +101,7 @@ define_pd_global(intx, InlineSmallCode,          1000);
   product(bool, UseRVA20U64, true, "Use RVA20U64 profile")                       \
   product(bool, UseRVC, false, "Use RVC instructions")                           \
   product(bool, UseRVA22U64, false, EXPERIMENTAL, "Use RVA22U64 profile")        \
+  product(bool, UseRVA23U64, false, EXPERIMENTAL, "Use RVA23U64 profile")        \
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/globals_riscv.hpp:104]: Can we group the three RV profile-related options together?

Comment (Hamlin-Li) [reply src/hotspot/cpu/riscv/globals_riscv.hpp:104]: Sure, will do

File: src/hotspot/cpu/riscv/vm_version_riscv.hpp (hunk 1 above)
Comment (RealFYang) [reply src/hotspot/cpu/riscv/vm_version_riscv.hpp:191]: Yeah, I think that's a mistake. I don't think it's safe to assume availability of optional extensions. We should only assume mandatory extensions in each profile.

Comment (Hamlin-Li) [reply src/hotspot/cpu/riscv/vm_version_riscv.hpp:191]: Sure, I will correct it, thanks for discussion.

**Review by RealFYang — APPROVED** [review]
Updated change LGTM. Thanks.

**Comment by Hamlin-Li** [comment]
Thanks @RealFYang for your quick response and review!

**Pull request closed without merging**